src/Hydrator/MoneyHydrator.php

```diff
@@ -20,7 +20,7 @@ class MoneyHydrator implements HydratorInterface
     public function extract($object)
     {
         return [
-            'amount' => $object->getAmount(),
+            'amount' => $object->getAmount() / 100,
```

Comment:
There are some currencies that have more than 100 units like INR, so dividing by 100 would result in incorrect values. 

Converting to the desired unit is a "view layer" issue. If you're using Zend\Form, that would be your view layer. If you're using API's or SPA, its recommended to return the units, so you convert it in javascript or in the client.

If you get the extracted array and place it into hydrate method, it should return the exactly the same object.